staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -693,6 +693,8 @@ type APIResource struct {
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
 	// shortNames is a list of suggested short names of the resource.
 	ShortNames []string `json:"shortNames,omitempty" protobuf:"bytes,5,rep,name=shortNames"`
+	// groupNames is a list of group aliases this resource belongs to (e.g. 'all')
+	GroupNames []string `json:"groupNames,omitempty" protobuf:"bytes,6,rep,name=groupNames"`
```

Comment:
> Other thought: What if we extend ShortName to just include such things?

I don't think that you want to accidentally end up with a category name since that will do weird things in the CLI.  `all/some-name` is unlikely to play nice in kubectl.  

That does raise some interesting name-squatting concerns.  Perhaps a CLI user should be forced to opt-in to particular categories in some way?